sapi/cli/php_cli.c

```diff
@@ -558,16 +558,19 @@ static void cli_register_file_handles(void)
 	php_stream_to_zval(s_err, &ec.value);
 
 	Z_CONSTANT_FLAGS(ic.value) = 0;
-	ic.name = zend_string_init_interned("STDIN", sizeof("STDIN")-1, 0);
-	zend_register_constant(&ic);
+	zend_string *stdin_name = zend_string_init("STDIN", sizeof("STDIN")-1, 0);
```

Comment:
I noticed that not using interned strings gets rid of the memleaks so now the number of memleaks is reduced from 4 to only 1 due to `PHP_CLI_PROCESS_TITLE`:
```
zend_register_bool_constant(
			ZEND_STRL("PHP_CLI_PROCESS_TITLE"),
			is_ps_title_available() == PS_TITLE_SUCCESS,
			0, 0);
```

Reply:
I've just realized that the difference between the rest of the constants and this one is that it's a non-persistent constant with a non-persistent interned `zend_string` while the rest of them are persistent.

Reply:
You're likely just getting less leaks reported because persistent/permanent strings aren't tracked by `emalloc`. You should have more consistent results by using `ASASN` along with the `--asan` flag passed to `run-tests.php`.

Reply:
Right, I don't think that will actually help to show with interned strings.